test/ctia/flows/crud_test.clj

```diff
@@ -122,3 +126,113 @@
+                                            :id "sighting-1"
+                                            :tlp "green"
+                                            :groups ["groups1"])
+                        "sighting-2" (assoc sighting-minimal
+                                            :title "sighting two"
+                                            :id "sighting-2"
+                                            :tlp "green"
+                                            :groups ["groups1"])
+                        "sighting-3" (assoc sighting-minimal
+                                            :title "sighting three"
+                                            :id "sighting-3"
+                                            :tlp "green"
+                                            :groups ["groups1"])})
+           ident {:login "login"
+                  :groups ["group1"]}
+           event-store (helpers/get-store app :event)
+           get-fn (fn [ids] (vals (select-keys @store ids)))
+           delete-fn (fn [ids]
+                       (into {}
+                             (map #(array-map
+                                        %
+                                        (not= @store
+                                              (swap! store dissoc %))))
+                             ids))
+           delete-flow (fn [expected]
+                         (let [entity-ids (seq (keys expected))
+                               res
+                               (flows.crud/delete-flow
+                                :entity-type :sighting
+                                :get-fn get-fn
+                                :delete-fn delete-fn
+                                :entity-ids entity-ids
+                                :long-id-fn identity
+                                :services services
+                                :identity (map->Identity ident))
+                               _ (Thread/sleep 1000)
+                               deleted-events? (into {}
+                                                     (map #(->> (search-events event-store
+                                                                              ident
+                                                                              %)
+                                                                first
+                                                                some?
+                                                                (array-map %)))
+                                                     entity-ids)]
+                           (is (= expected
+                                  (into {} res)))
+                           (is (= expected deleted-events?)
+                               (str "flow shall return " expected))))]
+       (delete-flow {"sighting-1" true
+                     "sighting-2" true})
+       (delete-flow {"sighting-3" true "missing" false})))))
```

Comment:
Another comment it would probably be a good idea to test in a concurrent environment.

Like receiving two delete-flow on a non disjoint set of entities. This way you should be able to test that.

```
(concurrently
   (delete-flow sightings-1 ... sightings-10)
   (delete-flow sightings-5 .. sighting-15)
   (delete-flow sightings-3 .. sighting-8)
   (delete-flow sightings-10 .. sighting-20))
```

would deletes from sighting-1 to sighting-20. Even if for some call not all the entities asked to be removed will be removed during this specific call.